### PR TITLE
front: fix punctual object position calculation

### LINKED ISSUE #4246: Wrong punctual object position calculation
### What happened?

When we click on a track to set the position of an object the computed offset is invalid. 

We can reproduce the bug with:

- Signals
- Detectors
- Buffer stops

### What did you expect to happen?

_No response_

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to the editor
2. Select a track section and note its length (the real one not the length of the geometry)
3. Create a buffer stop and put it at the very end of the previous track
4. Check the computed offset of the buffer stop. It doesn't match the noted length

The front should calculate offsets in proportion to track length.

*NOTE*: This should be the case for multiple objects even not only punctual objects.

### What operating system, browser and environment are you using?

- **Browser**: Firefox v115
- **OS**: Nixos v23.05
- **Env**: Local

### OSRD version (top right corner `Account` button > `Informations`)

2421cbdb7d8085f6a541b0b2ffc053f51e1a9988

### PR DESCRIPTION
**Pull request by Akctarus**
closes #4246 

- [x] Add the approximateDistanceWithEditoastData function to the PointEditionToolFactory when moving/clicking on the map
- [x] Add a custom field in the PointEditionLeftPanel schema for the 3 entities (bufferStop, detector and signal)

### CODE REVIEW

File: front/src/applications/editor/tools/pointEdition/components.tsx [hunk 1]
```diff
@@ -311,16 +320,17 @@ export const PointEditionLeftPanel: FC<{ type: EditoastType }> = <Entity extends
             typeof oldPosition === 'number' &&
             newPosition !== oldPosition
           ) {
-            const point = along(trackState.track, newPosition, { units: 'meters' });
+            const turfPosition =
+              (newPosition * (length(trackState.track) * 1000)) /
```
Comment (clarani) [inline front/src/applications/editor/tools/pointEdition/components.tsx]:
```suggestion
              (newPosition * length(trackState.track, { unit: 'meters' })) /
```

File: front/src/applications/editor/tools/pointEdition/tool-factory.tsx [hunk 2]
```diff
@@ -15,10 +15,17 @@ import { LAYER_TO_EDITOAST_DICT, LayerType } from '../types';
...
 import { getPointEditionLeftPanel, POINT_LAYER_ID, PointEditionMessages } from './components';
 import { PointEditionState } from './types';
-import { NULL_GEOMETRY, BufferStopEntity, DetectorEntity, SignalEntity } from '../../../../types';
+import {
+  NULL_GEOMETRY,
+  BufferStopEntity,
+  DetectorEntity,
+  SignalEntity,
+  TrackSectionEntity,
+} from '../../../../types';
```
Comment (clarani) [inline front/src/applications/editor/tools/pointEdition/tool-factory.tsx]:
```suggestion
} from 'types';
```

**Review by clarani — APPROVED** [review]
LGTM, tested ! Left 2 small comments :white_check_mark:

File: front/src/applications/editor/tools/pointEdition/tool-factory.tsx (hunk 2 above)
Comment (Akctarus) [reply front/src/applications/editor/tools/pointEdition/tool-factory.tsx]: Fixed in last commit !

File: front/src/applications/editor/tools/pointEdition/components.tsx (hunk 1 above)
Comment (Akctarus) [reply front/src/applications/editor/tools/pointEdition/components.tsx]: Ty ! fixed in last commit

**Pull request merged**